tests/acceptance/stepDefinitions/filesContext.js

```diff
@@ -37,27 +38,48 @@ When('the user enables the setting to view hidden folders on the webUI', functio
   return client.page.filesPage().showHiddenFiles()
 })
 
+When('the user deletes file/folder {string} using the webUI', function (element) {
+  return client.page.filesPage().deleteFile(element)
+})
+
+When('the user deletes the following elements using the webUI', function (table) {
+  for (const line of table.rows()) {
+    client.page.filesPage().deleteFile(line[0])
+    deletedElementsTable.push(line[0])
+  }
+  return client.page.filesPage()
+})
+
 Then(/there should be no files\/folders listed on the webUI/, function () {
   return client.page.filesPage().allFileRows(function (result) {
     client.assert.equal(result.value.length, 0)
   })
 })
 
-Then('folder {string} should be listed on the webUI', function (folder) {
+Then('file/folder {string} should be listed on the webUI', function (folder) {
   return client
     .page
     .filesPage()
     .waitForFileVisible(folder)
 })
 
-Then('folder {string} should not be listed on the webUI', function (folder) {
-  const filesPage = client.page.filesPage()
-  filesPage
-    .waitForElementVisible('@filesTable')
-  filesPage
-    .useXpath()
-    .waitForElementNotPresent(filesPage.getFileRowSelectorByFileName(folder))
-  return filesPage
+Then('file/folder {string} should not be listed on the webUI', function (folder) {
+  return client.page.filesPage().assertElementNotListed(folder)
+})
+
+Then('the deleted elements should not be listed on the webUI', function () {
+  for (const element of deletedElementsTable) {
+    client.page.filesPage().assertElementNotListed(element)
+  }
+  return client
+})
+
+Then('the deleted elements should not be listed on the webUI after a page reload', function () {
+  client.refresh()
+  for (const element of deletedElementsTable) {
```

Comment:
Can you call the method above that already does this loop?

Reply:
that should not be done see https://docs.cucumber.io/guides/anti-patterns/#support-for-conjunction-steps
and I also cannot see how that would work in JS, `Then()` is a function call and not a function
would have to make a outside function that is called from both places

Reply:
made a small helper function that is called from both steps